src/datasources/blockchain/blockchain-api.service.ts

```diff
@@ -0,0 +1,47 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { Chain as DomainChain } from '@/domain/chains/entities/chain.entity';
+import { RpcUriAuthentication } from '@/domain/chains/entities/rpc-uri-authentication.entity';
+import { IBlockchainApi } from '@/domain/interfaces/blockchain-api.interface';
+import { Injectable } from '@nestjs/common';
+import { Chain, PublicClient, createPublicClient, http } from 'viem';
+
+@Injectable()
+export class BlockchainApi implements IBlockchainApi {
+  private readonly infuraApiKey: string;
+
+  constructor(
+    private readonly configurationService: IConfigurationService,
+    private readonly chain: DomainChain,
+    public readonly destroyClient: (chainId: string) => void,
+  ) {
+    this.infuraApiKey = this.configurationService.getOrThrow<string>(
+      'blockchain.infura.apiKey',
+    );
+  }
+
+  getClient(): PublicClient {
+    return createPublicClient({
+      chain: this.getChain(),
+      transport: http(),
+    });
+  }
```

Comment:
I'm not totally against this, but wanted to known the approach better: why do we expose the `PublicClient` outside this class? Wouldn't it be more inline with the other datasource  classes (`*-api.service.ts`) to just expose the functions itself (even we are mostly wrapping the client in this case)?

I mean, instead of the having the clients of this class doing:
```
const blockchainClient = blockChainApi.getClient();
const blockNumber = await blockchainClient.getBlockNumber();
```
They would do something like:
```
const blockNumber = await blockchainApi.getBlockNumber();
```

Exposing the `PublicClient` seems a bit like defeating the purpose of having a datasource, which in my opinion has as one of its main benefits the encapsulation of the underlying implementation. What do you think?

Reply:
Good point. `IBlockchainApi` is essentially redundant (it was a remanant from trialling the implementation). I removed it in e38f10916a1a2c4fba22cc228849c2777a96d1c6 as per your suggestion.